src/compression.js

```diff
@@ -12,15 +12,40 @@ export function decideCompression(compressionSupport) {
     }
 }
 
+function hasMagicGzipHeader(compressionResultElement) {
+    try {
+        const a = compressionResultElement[0]
+        const b = compressionResultElement[1]
+        return a === 31 && b === 139
+    } catch (e) {
+        return false
+    }
+}
+
 export function compressData(compression, jsonData, options) {
     if (compression === 'lz64') {
         return [{ data: LZString.compressToBase64(jsonData), compression: 'lz64' }, options]
     } else if (compression === 'gzip-js') {
         // :TRICKY: This returns an UInt8Array. We don't encode this to a string - returning a blob will do this for us.
-        return [
+        const compressionResult = [
             gzipSync(strToU8(jsonData), { mtime: 0 }),
             { ...options, blob: true, urlQueryArgs: { compression: 'gzip-js' } },
         ]
+
+        // temporary logging to identify source of https://github.com/PostHog/posthog/issues/4816
+        if (
+            !jsonData ||
+            jsonData === 'undefined' ||
+            !compressionResult[0] ||
+            !hasMagicGzipHeader(compressionResult[0])
+        ) {
+            posthog.capture('PostHogJSCompressionCannotBeDecompressed', {
```

Comment:
This will send the result back to the _users_ posthog project, which is not what you want here.

Check out src/capture-metrics.js instead for example.

You also want to make sure we only log when debug config is enabled.